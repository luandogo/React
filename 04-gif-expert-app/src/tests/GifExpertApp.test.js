import React from 'react';

import { shallow } from "enzyme";
import { GifExpertApp } from "../GifExpertApp";



describe('Tests in <GifExpertApp />', () => {


    test('should be displayed correctly', () => {
        const wrapper = shallow( <GifExpertApp/> );
        expect( wrapper ).toMatchSnapshot();
        
    }); 

    test('should display a list of categories.', () => {
        const categories = ['One Piece', 'One Punch'];
        const wrapper = shallow( <GifExpertApp defaultCategories={ categories }/> );

        expect( wrapper ).toMatchSnapshot();
        expect( wrapper.find('GifGrid').length ).toBe( categories.length );
        
    })
    
});