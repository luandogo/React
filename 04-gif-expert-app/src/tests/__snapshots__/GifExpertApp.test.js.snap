// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Tests in <GifExpertApp /> should be displayed correctly 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol />
</Fragment>
`;

exports[`Tests in <GifExpertApp /> should display a list of categories. 1`] = `
<Fragment>
  <h2>
    GifExpertApp
  </h2>
  <AddCategory
    setCategories={[Function]}
  />
  <hr />
  <ol>
    <GifGrid
      category="One Piece"
      key="One Piece"
    />
    <GifGrid
      category="One Punch"
      key="One Punch"
    />
  </ol>
</Fragment>
`;
